src/components/NewClusterModal.js

```diff
@@ -227,18 +290,20 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
           disabled: isSelectedImageInputted && isCustomImageInvalid,
           tooltip: isSelectedImageInputted && isCustomImageInvalid && 'Enter a valid docker image to use',
           onClick: () => {
-            if (isSelectedImageInputted) {
+            if (isSelectedImageInputted && !this.canUpdate()) {
               this.setState({ viewMode: 'warning' })
             } else if (!!currentCluster) {
-              this.setState({ viewMode: 'replace' })
+              this.setState({ viewMode: _.lowerCase(getUpdateOrReplace()) })
```

Comment:
Nit: This would be more consistent with line 294 if `getUpdateOrReplace` returns lowercase symbols, followed by using `_.startCase` to render the button text.